### PR TITLE
Closes #2683. Update Gradle to `5.5`

### LINKED ISSUE #2683: Update Gradle to latest (5.+)
Hi. Currently project is building with Gradle `4.10`. The latest one is `5.4.1`.
There are [bunch of improvements](https://gradle.org/whats-new/gradle-5/) in Gradle `5.0`
(e.g. speed, build caching, incremental annotation processing, memory efficiency).

I'm gonna investigate this and upgradle to the latest Gradle. Are there some major issues why we can't use the newest version that I need to know in advance?

### PR DESCRIPTION
**Pull request by dector**
### Issue: #2683.

Updated Gradle to `5.4.1`.

[What's new in Gradle 5](https://gradle.org/whats-new/gradle-5/)
[Release notes for 5.4.1](https://docs.gradle.org/5.4.1/release-notes.html)

### Pull Request checklist
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] ~~**Tests**: This PR includes thorough tests or an explanation of why it does not~~
- [x] ~~**Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one~~
- [x] ~~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features~~

### CODE REVIEW

**Comment by dector** [comment]
I have 404 for [bitrise report](https://app.bitrise.io/build/9aab41abc5b4795a). :(

File: buildSrc/src/main/java/Dependencies.kt [hunk 1]
```diff
@@ -26,6 +26,7 @@ private object Versions {
     const val androidx_lifecycle_savedstate = "1.0.0-alpha01"
     const val androidx_testing = "1.1.0-alpha08"
     const val androidx_core = "1.2.0-alpha01"
+    const val androidx_paging = "2.0.0"
```
Comment (dector) [inline buildSrc/src/main/java/Dependencies.kt]: Newest one is `2.1.0` but I used `2.0.0` here because it was version used from transitive dependency.

**Comment by sarentz** [comment]
Thank you for this contribution @dector - We definitely want to upgrade Gradle, but we have decided to not land this right now. The reason is that we are getting very close to a release and we don't want to introduce regressions or unexpected things in the build system. So we will stick with 4.x a little longer and then land your patch probably in the beginning of July.

**Comment by colintheshots** [comment]
Since I wasn't able to push to the branch, I created a new PR from this branch, updating to Gradle 5.5: https://github.com/mozilla-mobile/fenix/pull/3929

**Pull request closed without merging**